### PR TITLE
Enhanced error reporting and security services

### PR DESCRIPTION
**Pull request by bistline**
* Adding Sentry & TCell integration
* Updating Rails & dependencies
* Adding new environment variables to support new integrations & other deployments

### CODE REVIEW

File: bin/boot_docker [hunk 1]
```diff
@@ -25,6 +26,10 @@ $0 [OPTION]
 -P VALUE	set the SENDGRID_PASSWORD variable (used for emails)
 -o VALUE	set the OAUTH_CLIENT_ID variable (for Google OAuth2)
 -S VALUE	set the OAUTH_CLIENT_SECRET variable (for Google OAuth2)
+-y VALUE	set the SENTRY_DSN variable (for Sentry error reporting, no default)
+-g VALUE  set the GA_TRACKING_ID variable for tracking usage via Google Analytics (no default)
+-w VALUE  set the TCELL_APP_ID variable (for integrating with TCell web application firewall, if enabled)
+-W VALUE  set the TCELL_API_KEY variable (for integrating with TCell web application firewall, if enabled)
```
Comment (eweitz) [inline bin/boot_docker]: Whitespace inconsistency here -- it seems spaces were used instead of tabs.

File: bin/boot_docker [hunk 2]
```diff
@@ -95,6 +103,18 @@ case $OPTION in
...
+		;;
+	g)
+	  GA_TRACKING_ID="$OPTARG"
+	  ;;
+  w)
+    TCELL_APP_ID="$OPTARG"
+    ;;
+  W)
+    TCELL_API_KEY="$OPTARG"
+    ;;
```
Comment (eweitz) [inline bin/boot_docker]: Ensure whitespace is consistent.

File: bin/run_unit_tests.sh [hunk 3]
```diff
@@ -35,22 +37,36 @@ then
...
-	bundle exec ruby -I test test/integration/taxons_controller_test.rb
-	bundle exec ruby -I test test/api/studies_controller_test.rb
-	bundle exec ruby -I test test/api/study_files_controller_test.rb
-	bundle exec ruby -I test test/api/study_file_bundles_controller_test.rb
-	bundle exec ruby -I test test/api/study_shares_controller_test.rb
-	bundle exec ruby -I test test/api/directory_listings_controller_test.rb
-	bundle exec ruby -I test test/models/cluster_group_test.rb
-	bundle exec ruby -I test test/models/user_annotation_test.rb
-	bundle exec ruby -I test test/models/parse_utils_test.rb
+  declare -a tests=(test/integration/fire_cloud_client_test.rb
```
Comment (eweitz) [inline bin/run_unit_tests.sh]: Nice refinement in this block!

File: README.md [hunk 4]
```diff
@@ -405,7 +464,7 @@ and any differing values for hostnames/client secrets/passwords as needed.
 The Single Cell Portal stores uploaded study data files in [FireCloud](https://software.broadinstitute.org/firecloud/) 
 workspaces, which in turn store data in GCP buckets.  This is all managed through a GCP service account which in turn 
 owns all portal workspaces and manages them on behalf of portal users.  All portal-related workspaces are within the 
-`single-cell-portal` namespace, which should be noted is a separate project from the one the portal operates out of.
+`single-cell-portal` namespace (or ), which should be noted is a separate project from the one the portal operates out of.
```
[the remark this answers is not in the document]
Comment (bistline) [reply README.md]: that's not how it should read... updating

File: bin/boot_docker (hunk 1 above)
Comment (bistline) [reply bin/boot_docker]: I'll check to see what the majority are and conform to that.

**Pull request merged**